ec/provider_config.go

```diff
@@ -33,22 +35,36 @@ const (
 )
 
 // configureAPI implements schema.ConfigureFunc
-func configureAPI(d *schema.ResourceData) (interface{}, error) {
+func configureAPI(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
 	timeout, err := time.ParseDuration(d.Get("timeout").(string))
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
 	if err != nil {
-		return nil, err
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to get EC client",
+			Detail:   "Invalid timeout value for EC client",
+		})
+		return nil, diags
 	}
 
 	authWriter, err := auth.NewAuthWriter(auth.Config{
 		APIKey:   d.Get("apikey").(string),
 		Username: d.Get("username").(string),
 		Password: d.Get("password").(string),
 	})
+
 	if err != nil {
-		return nil, err
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to get EC client",
+			Detail:   "Invalid credentials for EC client",
+		})
+		return nil, diags
```

Comment:
```suggestion
		return nil, diag.FromErr(err)
```

Reply:
Thanks @karencfv for the review 😄 have addressed the suggested changes.